last-rev-llc/contentful-scripts
# IFWEB-829 - Add includedLocales field to all page content types

### Description
Added script to add includedLocales field and then update that field

### Jira Ticket
[IFWEB-829](https://lastrev.atlassian.net/browse/IFWEB-829)

### Closes issue(s)
[IFWEB-829](https://lastrev.atlassian.net/browse/IFWEB-829)

### How to test
Go to [contentful](https://app.contentful.com/spaces/hhv516v5f7sj/settings/environments) and create another environment (copy from master)

On local machine:

Add Field:
Change contentful environment variable in the .env file to the name of the environment you just created above
Go to /migrations/impossible/IFWEB-829-includedLocales.js
Make sure updateFields() method is commented out on line 84
Open terminal and run `npm run migration`
Go to contentful and see that for each page type (`aaShopifyProduct`, `pageBlog`, `pageGeneral`, `pageNews`, `pagePerson`, `pageRecipe`) the Included Locales field has been added and placed below the slug field

Update Fields:
Now go back to /migrations/impossible/IFWEB-829-includedLocales.js
Uncomment line 84 and comment out addFields() method on line 56
Open terminal and run `npm run migration`
Any page that was in draft should have all locales selected
Any page that was published should only have `en-US`, `ca`, `hk`, `hk-en` selected


### Preview URL
[Deploy Preview URL](https://example.com)

### Screenshots (if applicable)
![Screen Shot 2021-08-18 at 5 56 02 PM](https://user-images.githubusercontent.com/51485419/129977513-8024304b-2746-41a0-8f4d-696492e04e7e.png)
![Screen Shot 2021-08-18 at 5 56 45 PM](https://user-images.githubusercontent.com/51485419/129977598-511d9331-f248-4550-a93d-13080f662da6.png)
![Screen Shot 2021-08-18 at 5 57 47 PM](https://user-images.githubusercontent.com/51485419/129977691-bef8f0bd-b7ea-4166-9420-3fd4dc0f194a.png)
![Screen Shot 2021-08-18 at 5 58 20 PM](https://user-images.githubusercontent.com/51485419/129977745-677e1690-3b52-4c7f-bb04-dcf2fb66bef7.png)


### Changes include
- [x] New Function
